File: 25/09/3005.go
package main

import (
	"fmt"
)

func maxFrequencyElements(nums []int) int {
	m := make(map[int]int)
	ans := 0
	max := 0
	for _, x := range nums {
		m[x]++
		c := m[x]
		if c > max {
			max = c
			ans = c
		} else if c == max {
			ans += c
		}
	}
	return ans
}

func main() {
	// 示例测试
	nums := []int{1, 2, 2, 3, 1, 4}
	fmt.Println(maxFrequencyElements(nums)) // 输出: 4
}

File: 25/09/3005.cpp
#include <vector>
#include <iostream>
#include <unordered_map>

int maxFrequencyElements(std::vector<int> &nums)
{
    std::unordered_map<int,int> m;
    std::for_each(nums.begin(),nums.end(),[&m](int x){
        m[x]++;
    });
    int max = 0;
    for_each(m.begin(),m.end(),[&max](std::pair<int,int> p){
        if (p.second > max) max = p.second;
    });
    int count = 0;
    std::for_each(m.begin(),m.end(),[&count,max](std::pair<int,int> p){
        if (p.second == max) count++;
    });

    return count;
}

int enhance_1(std::vector<int> nums) {
    std::unordered_map<int,int> m;
    int ans = 0, max = 0;
    std::for_each(nums.begin(),nums.end(),[&m,&ans,&max](int x) {
        int c = ++m[x];
        if (c > max) {
            ans = max = c;
        }else if (c == max) {
            ans += c;
        }
    });
    return ans;
}